### PR TITLE
Add two quantum embeddings

### LINKED ISSUE #2982: Add common quantum embeddings
### Required prerequisites

- [x] Search the [issue tracker](https://github.com/NVIDIA/cuda-quantum/issues) to check if your feature has already been mentioned or rejected in other issues.

### Describe the feature

Add built-in APIs for common quantum embeddings.

--

**(1) Amplitude encoding:**
Maps and normalizes classical features (`list`, `ndarray` or `cudaq.State`) to a quantum state by encoding values as amplitudes:  $x \mapsto \sum_i x_i \|i\rangle$, including `pad` as an argument for vectors not sized to $2^n$ with the specified pad value, e.g., zero-padding to the nearest power of $2^n$ by setting `pad = 0`. The featured API looks like:
```
state_vector = cudaq.amplitude_encode([0.5, 0.5, 0.5], pad = 0)
```
Returns `cudaq.State` with values `[0.5773, 0.5773, 0.5773, 0]` (L2-norm = 1)

--

**(2) Angular encoding:**
Encodes classical features (`list` or `ndarray`) as qubit rotations (e.g., with specified Ry gates):  $x_i \mapsto R_y(x_i)\|0\rangle$:
```
@cudaq.kernel
def kernel():
    q = cudaq.amplitude_encode([0.1, 0.2, 0.3], rotation='Y')
```
The corresponding kernel:
```
print(cudaq.draw(kernel))
     ╭─────────╮
q0 : ┤ ry(0.1) ├
     ├─────────┤
q1 : ┤ ry(0.2) ├
     ├─────────┤
q2 : ┤ ry(0.3) ├
     ╰─────────╯
``` 

--

Both embedding methods are commonly used in QML, enabling developers to design models more easily and intuitively.

### PR DESCRIPTION
**Pull request by simsaidan**
Resolves #2982

Implements the amplitude encoding and angular encoding. Supports both numpy and cupy ndarrays. For the second encoding, blocks a non-kernel call with an error. Updates the ast bridge, which does some compile time checks.
Used cursor for planning.

### CODE REVIEW

**Comment by 1tnguyen** [comment]
> /ok to test d3c221b77c86edf3fe14e1b3cc59c437316c016d

**Command Bot:** Processing...

**Comment by schweitzpgi** [comment]
C++ implementation?

**Comment by schweitzpgi** [comment]
Is this a dup of #4666 ?

**Comment by simsaidan** [comment]
Hi @schweitzpgi @1tnguyen I extended the implementation over to C++.

**Comment by bettinaheim** [comment]
@simsaidan  Thank you for this contribution!
- Can you please update the PR to have this go into the contrib module/contrib namespace?
- I'll re-trigger CI; please keep an eye out and get this into a state where all pipelines pass. 
- We'll help to unify this contribution with #4666, which contains additional Python tests for builder mode and nice doc comments. 
We appreciate if you are willing to help bring this across the finish line in a collaborative manner.

**Comment by bettinaheim** [comment]
> /ok to test ba96e03

**Command Bot:** Processing...

**Comment by sacpis** [comment]
> /ok to test 44e0ef5

**Command Bot:** Processing...

**Comment by simsaidan** [comment]
Hey @bettinaheim ! I updated the PR and moved the embeddings to the contrib namespace. I improved the doc comments and added support for builder mode.

**Comment by simsaidan** [comment]
Hey @bettinaheim @schweitzpgi ! Could you trigger the CI check again so I can see if it is green or otherwise what I need to fix?

**Comment by sacpis** [comment]
@simsaidan Have you built the code and ran the tests locally?

**Comment by simsaidan** [comment]
@sacpis Yes, I ran CI and tests locally and no fails

**Comment by sacpis** [comment]
For the latest spelling fix [7a52c2e](https://github.com/NVIDIA/cuda-quantum/pull/4667/commits/7a52c2e2d31fefd3b368ab09bf8229fa22b544f8), please use backticks around those words as those are not english words to be added to `.github/pre-commit/spelling_allowlist.txt`.

**Comment by sacpis** [comment]
Also, would you please sign your commits?

**Comment by sacpis** [comment]
@simsaidan I am seeing the following build failure locally

```
[896/1343] Building CXX object unittests/CMakeFiles/test_runtime_qpp.dir/integration/kernels_tester.cpp.o
FAILED: unittests/CMakeFiles/test_runtime_qpp.dir/integration/kernels_tester.cpp.o 
ccache /usr/local/toolchain/bin/c++ -DCUDAQ_LIBRARY_MODE -DCUDAQ_SIMULATION_SCALAR_FP64 -DFMT_HEADER_ONLY=1 -DHAS_OPENMP -DNVQIR_BACKEND_NAME=qpp -DQPP_VERSION_NUM=3.1 -DQPP_VERSION_STR=\"3.1\" -D__MATH_LONG_DOUBLE_CONSTANTS -Dtest_runtime_qpp_EXPORTS -I/workspaces/cuda-quantum/build/cudaq/include -I/workspaces/cuda-quantum/cudaq/include -I/workspaces/cuda-quantum/unittests/. -I/workspaces/cuda-quantum/runtime/nvqir/qpp -I/workspaces/cuda-quantum/runtime -I/workspaces/cuda-quantum/tpls/eigen -I/workspaces/cuda-quantum/tpls/fmt/include -I/workspaces/cuda-quantum/runtime/include -I/usr/local/cuda-12.6/targets/x86_64-linux/include -I/workspaces/cuda-quantum/runtime/internal/compiler/include -isystem /usr/local/llvm/include -isystem /workspaces/cuda-quantum/tpls/json/single_include -isystem /workspaces/cuda-quantum/tpls/json/single_include_fwd -isystem /workspaces/cuda-quantum/tpls/qpp/include -isystem /workspaces/cuda-quantum/tpls/qpp/qasmtools/include -isystem /workspaces/cuda-quantum/tpls/googletest-src/googletest/include -isystem /workspaces/cuda-quantum/tpls/googletest-src/googletest -Wno-attributes -Wno-ctad-maybe-unsupported -O3 -DNDEBUG -std=gnu++20 -fPIE   -D_GNU_SOURCE -D_DEBUG -D_GLIBCXX_ASSERTIONS -D_GLIBCXX_USE_CXX11_ABI=1 -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -Wno-error=deprecated-declarations -Wno-uninitialized-const-pointer -Wno-restrict -Wno-stringop-overflow -Wno-stringop-overread --param=evrp-mode=legacy -Wno-character-conversion -Werror -MD -MT unittests/CMakeFiles/test_runtime_qpp.dir/integration/kernels_tester.cpp.o -MF unittests/CMakeFiles/test_runtime_qpp.dir/integration/kernels_tester.cpp.o.d -o unittests/CMakeFiles/test_runtime_qpp.dir/integration/kernels_tester.cpp.o -c /workspaces/cuda-quantum/unittests/integration/kernels_tester.cpp
In file included from ../unittests/integration/kernels_tester.cpp:10:
../runtime/cudaq/builder/kernels.h: In instantiation of ‘void cudaq::contrib::angular_encode(Kernel&&, cudaq::QuakeValue&, cudaq::QuakeValue&, RotationAxis) [with Kernel = cudaq::kernel_builder<std::vector<double, std::allocator<double> > >&]’:
../unittests/integration/kernels_tester.cpp:250:33:   required from here
../runtime/cudaq/builder/kernels.h:238:30: error: no matching function for call to ‘applyAxisRotation(cudaq::kernel_builder<std::vector<double, std::allocator<double> > >&, cudaq::contrib::RotationAxis&, cudaq::QuakeValue, cudaq::QuakeValue)’
  238 |     detail::applyAxisRotation(kernel, rotation, angles[i], q[i]);
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../runtime/cudaq/builder/kernels.h:163:6: note: candidate: ‘template<class Kernel> void cudaq::contrib::detail::applyAxisRotation(Kernel&, cudaq::contrib::RotationAxis, cudaq::QuakeValue&, cudaq::QuakeValue&)’
  163 | void applyAxisRotation(Kernel &kernel, RotationAxis axis, QuakeValue &theta,
      |      ^~~~~~~~~~~~~~~~~
../runtime/cudaq/builder/kernels.h:163:6: note:   template argument deduction/substitution failed:
../runtime/cudaq/builder/kernels.h:238:30: note:   cannot convert ‘cudaq::QuakeValue::operator[](const cudaq::QuakeValue&)((*(const cudaq::QuakeValue*)(& i)))’ (type ‘cudaq::QuakeValue’) to type ‘cudaq::QuakeValue&’
  238 |     detail::applyAxisRotation(kernel, rotation, angles[i], q[i]);
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../runtime/cudaq/builder/kernels.h:179:6: note: candidate: ‘template<class Kernel> void cudaq::contrib::detail::applyAxisRotation(Kernel&, cudaq::contrib::RotationAxis, double, cudaq::QuakeValue&)’
  179 | void applyAxisRotation(Kernel &kernel, RotationAxis axis, double theta,
      |      ^~~~~~~~~~~~~~~~~
../runtime/cudaq/builder/kernels.h:179:6: note:   template argument deduction/substitution failed:
../runtime/cudaq/builder/kernels.h:238:30: note:   cannot convert ‘cudaq::QuakeValue::operator[](const cudaq::QuakeValue&)((*(const cudaq::QuakeValue*)(& i)))’ (type ‘cudaq::QuakeValue’) to type ‘double’
  238 |     detail::applyAxisRotation(kernel, rotation, angles[i], q[i]);
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../runtime/cudaq/builder/kernels.h: In instantiation of ‘void cudaq::contrib::angular_encode(Kernel&&, cudaq::QuakeValue&, std::span<const double>, RotationAxis) [with Kernel = cudaq::kernel_builder<>&]’:
../runtime/cudaq/builder/kernels.h:262:17:   required from ‘void cudaq::contrib::angular_encode(Kernel&&, cudaq::QuakeValue&, const std::vector<double>&, RotationAxis) [with Kernel = cudaq::kernel_builder<>&]’
../unittests/integration/kernels_tester.cpp:228:33:   required from here
../runtime/cudaq/builder/kernels.h:254:30: error: no matching function for call to ‘applyAxisRotation(cudaq::kernel_builder<>&, cudaq::contrib::RotationAxis&, std::span<const double>::element_type&, cudaq::QuakeValue)’
  254 |     detail::applyAxisRotation(kernel, rotation, angles[i], q[i]);
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../runtime/cudaq/builder/kernels.h:163:6: note: candidate: ‘template<class Kernel> void cudaq::contrib::detail::applyAxisRotation(Kernel&, cudaq::contrib::RotationAxis, cudaq::QuakeValue&, cudaq::QuakeValue&)’
  163 | void applyAxisRotation(Kernel &kernel, RotationAxis axis, QuakeValue &theta,
      |      ^~~~~~~~~~~~~~~~~
../runtime/cudaq/builder/kernels.h:163:6: note:   template argument deduction/substitution failed:
../runtime/cudaq/builder/kernels.h:254:30: note:   cannot convert ‘angles.std::span<const double>::operator[](i)’ (type ‘std::span<const double>::element_type’ {aka ‘const double’}) to type ‘cudaq::QuakeValue&’
  254 |     detail::applyAxisRotation(kernel, rotation, angles[i], q[i]);
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../runtime/cudaq/builder/kernels.h:179:6: note: candidate: ‘template<class Kernel> void cudaq::contrib::detail::applyAxisRotation(Kernel&, cudaq::contrib::RotationAxis, double, cudaq::QuakeValue&)’
  179 | void applyAxisRotation(Kernel &kernel, RotationAxis axis, double theta,
      |      ^~~~~~~~~~~~~~~~~
../runtime/cudaq/builder/kernels.h:179:6: note:   template argument deduction/substitution failed:
../runtime/cudaq/builder/kernels.h:254:30: note:   cannot convert ‘cudaq::QuakeValue::operator[](std::size_t)(i)’ (type ‘cudaq::QuakeValue’) to type ‘cudaq::QuakeValue&’
  254 |     detail::applyAxisRotation(kernel, rotation, angles[i], q[i]);
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1plus: note: unrecognized command-line option ‘-Wno-character-conversion’ may have been intended to silence earlier diagnostics
cc1plus: note: unrecognized command-line option ‘-Wno-uninitialized-const-pointer’ may have been intended to silence earlier diagnostics
[929/1343] Building CXX object unittests/CMakeFiles/test_runtime_qpp.dir/backends/QPPTester.cpp.o
ninja: build stopped: subcommand failed.
```

**Comment by simsaidan** [comment]
@sacpis I added backticks and removed those words from the allowlist. I also made a change to address that build failure. I signed the latest commit. Do I need to anything retroactively to the commits before or does signing the latest one cover me?

File: python/cudaq/contrib/encoding.py [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+        "amplitude_encode: data must be a list, 1D array, or cudaq.State.")
+
+
+@trace.traced
+def amplitude_encode(
+    data: list | np.ndarray | State,
+    *,
+    pad: complex | float = 0,
+) -> State:
+    """
```
Comment (sacpis) [inline python/cudaq/contrib/encoding.py]: Let's make these docstrings raw in order to avoid `\r` escape. 
Would you please run `scripts/build_docs.sh` and see if these are rendered as expected?

Comment (sacpis) [reply python/cudaq/contrib/encoding.py]: Please check all symbols followed by `\`.

File: runtime/cudaq/algorithms/encoding.cpp [hunk 2]
```diff
@@ -0,0 +1,133 @@
+/*******************************************************************************
+ * Copyright (c) 2026 NVIDIA Corporation & Affiliates.  * All rights reserved.
+ *                                                      *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
```
Comment (sacpis) [inline runtime/cudaq/algorithms/encoding.cpp:7]: Let's fix the header here.

File: runtime/cudaq/algorithms/encoding.h [hunk 3]
```diff
@@ -0,0 +1,74 @@
+/****************************************************************-*- C++ -*-****
+ * Copyright (c) 2026 NVIDIA Corporation & Affiliates.  * All rights reserved.
+ *                                                      *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
```
Comment (sacpis) [inline runtime/cudaq/algorithms/encoding.h:7]: Let's fix the header here.

File: unittests/nvqpp/integration/encoding_tester.cpp [hunk 4]
```diff
@@ -0,0 +1,65 @@
+/*******************************************************************************
+ * Copyright (c) 2026 NVIDIA Corporation & Affiliates.  * All rights reserved.
+ *                                                      *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
```
Comment (sacpis) [inline unittests/nvqpp/integration/encoding_tester.cpp:7]: Let's fix the header here.

**Review by sacpis — COMMENTED** [review]
Thanks @simsaidan. Have put some comments. Let's run the CI now.

**Comment by sacpis** [comment]
> /ok to test f905c18

**Command Bot:** Processing...

**Comment by sacpis** [comment]
Would you please add the following text in an empty commit? It will sign the commits which does not contain your signature.

```
DCO Remediation Commit for Aidan Sims <ans248@cornell.edu>

I, Aidan Sims <ans248@cornell.edu>, hereby add my Signed-off-by to this commit: 76a7a1a8a6b33c549e1df8f37bf9c64a0ebf96c5
I, Aidan Sims <ans248@cornell.edu>, hereby add my Signed-off-by to this commit: 7a52c2e2d31fefd3b368ab09bf8229fa22b544f8

Signed-off-by: Aidan Sims <ans248@cornell.edu>
```

Please do run this command to commit it

```
git commit -s --allow-empty
```

**Comment by simsaidan** [comment]
@sacpis @bettinaheim Added the empty commit and fixed the headers and docstrings. I also made an edit which I hope will address the latest CI failures

File: python/cudaq/contrib/encoding.py [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+    Returns:
+      :class:`State`: Normalized state vector suitable for simulation and
+          ``cudaq.State.from_data`` workflows.
+
+    Raises:
+      ValueError: If ``data`` is empty, has zero norm after padding, or is not
+          a 1D vector.
+      TypeError: If ``data`` has an unsupported type.
+
+    # Example:
```
Comment (sacpis) [inline python/cudaq/contrib/encoding.py]: I think we can put examples under `examples/` as these are not rendered properly in the API section.

Image

**Comment by sacpis** [comment]
> /ok to test 95c8527

**Command Bot:** Processing...

**Comment by sacpis** [comment]
Please run `bash scripts/run_all_spelling.sh` before pushing the fix for spellings.

**Comment by sacpis** [comment]
Seems like a regression caused with the latest changes.
https://github.com/NVIDIA/cuda-quantum/actions/runs/27234942193/job/80431397976?pr=4667#step:13:2914

**Comment by simsaidan** [comment]
@sacpis Spelling should be good now and I moved the examples. For the regression, the fix for the last error, I added a conversion step between sized and unsized registers. That change makes the past errors pass but makes this new test fail. Should I try to find some other workaround that would make both pass?

**Comment by sacpis** [comment]
> @sacpis Spelling should be good now and I moved the examples. For the regression, the fix for the last error, I added a conversion step between sized and unsized registers. That change makes the past errors pass but makes this new test fail. Should I try to find some other workaround that would make both pass?

Thanks @simsaidan. We would like to see everything passing with this PR.

**Comment by sacpis** [comment]
Also, would you please collaborate with @ssmswapnil as per @bettinaheim's comment [here](https://github.com/NVIDIA/cuda-quantum/pull/4666#issuecomment-4623648700) to unify with PR #4666?

**Comment by simsaidan** [comment]
Hey @ssmswapnil! Nice to meet you. The maintainers suggested we coordinate on merging our work. I've previously incorporated the builder mode support and docstring improvements that were in your PR. Could you take a look and see if there's anything else from your implementation that we should bring over?

**Comment by ssmswapnil** [comment]
Hey @simsaidan, nice to meet you too! Thanks for incorporating those. I'll go through your updated PR and check what else might be useful to bring over. Will get back to you shortly.

**Comment by ssmswapnil** [comment]
Hey @simsaidan, I've gone through your PR — the C++ support and AST bridge work looks great. From my side, the main things I had were:

1. 20 unit tests covering edge cases (empty input, zero vectors, padding, complex inputs, invalid rotations, parameterized kernels)
2. `_LAZY_ATTRS` registration in `__init__.py` for top-level access

Do you already have those covered, or would it help if I ported them over?

**Comment by simsaidan** [comment]
@ssmswapnil Thanks! That makes a lot of sense and I added the lazy attrs registration and some tests you had but I did not. 

@sacpis @bettinaheim I also made a change which might make both opposing tests pass for CI but not 100% sure.

**Comment by sacpis** [comment]
> /ok to test 4f26d8d

**Command Bot:** Processing...

**Comment by ssmswapnil** [comment]
@simsaidan , glad it helped! Let me know if there's anything else I can do to help get this across the finish line.

**Comment by simsaidan** [comment]
@sacpis @bettinaheim Added one more commit to fix the two failures in the last CI run. Could we run it again? Thank you for your patience as I work to get CI green.

**Comment by sacpis** [comment]
> /ok to test 8cacdc4

**Command Bot:** Processing...

**Comment by sacpis** [comment]
> /ok to test 11df846

**Command Bot:** Processing...

**Comment by simsaidan** [comment]
@sacpis Could we run the CI again? Pre-commit is green and I’ve run the Python tests locally (test_encoding.py, test_angular_encode.py, lazy-import tests). I'm having some local trouble running the C++ build, so if this run fails, I will look to rebuild from scratch.

**Comment by sacpis** [comment]
@simsaidan Thank you. May I please know how are you building CUDA-Q locally? And running the tests locally?

**Comment by simsaidan** [comment]
@sacpis On macOS I’m running Python tests and pre-commit against an installed CUDA-Q build with PYTHONPATH pointed at my checkout. I attempted a native source build but my local C++ toolchain is currently broken, and I don’t have the devcontainer running yet. So for this PR, I was trying to validate via CI, but I think its best to work toward the Docker/devcontainer setup.

**Comment by simsaidan** [comment]
@sacpis @bettinaheim Thanks for your patience on the CI iterations. I rebuilt the package from scratch locally with the arm64 toolchain. Now, I was able to reproduce the CI failures locally. I am pushing new code which makes that last test pass. I ran the PR-related C++ tests locally: all EncodingTester and KernelsTester.checkAngularEncode* tests pass.
I also ran the PR-related Python tests and the pre-commit and looked green.

If we could run the CI again here, I'd appreciate it.

**Comment by sacpis** [comment]
> /ok to test 92c2bac

**Command Bot:** Processing...

**Comment by simsaidan** [comment]
@sacpis CI was successful on the last run. However, I had two merge conflicts. I rebased and resolved the conflicts. Pytest, ctest, and the precommit are all green locally.

**Comment by sacpis** [comment]
Seems like DCO is still appearing as failing.

**Comment by sacpis** [comment]
> /ok to test bb0df9c

**Command Bot:** Processing...

**Comment by simsaidan** [comment]
@sacpis Oh I didnt realize the rebase removed the remediation. I fixed those two commits with the signature.

**Comment by sacpis** [comment]
> /ok to test 600df7a

**Command Bot:** Processing...

**Comment by sacpis** [comment]
> /ok to test c4da821

**Command Bot:** Processing...

**Comment by simsaidan** [comment]
@sacpis the CI run looks green.

**Review by sacpis — APPROVED** [review]
LGTM. Thanks @simsaidan and @ssmswapnil for your contribution to CUDA-Q. Appreciate fixing the tests in the PR.

**Comment by sacpis** [comment]
> /ok to test e677383

**Command Bot:** Processing...

**Comment by sacpis** [comment]
@simsaidan Would you please check and resolve the conversations?

**Comment by sacpis** [comment]
Also, would you please run `pre-commit run --all-files` to fix the formatting?

**Review by 1tnguyen — APPROVED** [review]
LGTM 👍 Thanks for the contributions!

**Comment by simsaidan** [comment]
@sacpis I resolved the comments and fixed the precommit issue

**Comment by sacpis** [comment]
> /ok to test c7af483

**Command Bot:** Processing...

**Comment by simsaidan** [comment]
@sacpis @1tnguyen I see the same two tests are failing in the deploy even through CI passed. I added a new commit which should fix that.

**Comment by 1tnguyen** [comment]
> /ok to test ca6f648

**Command Bot:** Processing...

**Comment by simsaidan** [comment]
@sacpis @1tnguyen I see you guys have added this to the merge queue many times, does that mean that the failing deploy tests are flaky? I looked at the error logs, but they do not offer much color as to what is going wrong.

**Comment by sacpis** [comment]
> /ok to test cc0bf71

**Command Bot:** Processing...

**Comment by simsaidan** [comment]
@sacpis Do you know why the merge keeps failing? I see the test issue but I’m confused why we keep adding it to the queue

**Comment by mitchdz** [comment]
I see 2 failures from the latest run:
* https://github.com/NVIDIA/cuda-quantum/actions/runs/27628267001/job/81696930204#step:21:41
```
Error: signing [ghcr.io/nvidia/cuda-quantum-dev:amd64-gcc12-pr-4667@sha256:ccb0c0d9a3dda424c4728ea08af93b436c44916d5c37e368490f73297daf83c7]: getting signer: getting key from Fulcio: fetching ambient OIDC credentials: invalid character 'u' looking for beginning of value
main.go:74: error during command execution: signing [ghcr.io/nvidia/cuda-quantum-dev:amd64-gcc12-pr-4667@sha256:ccb0c0d9a3dda424c4728ea08af93b436c44916d5c37e368490f73297daf83c7]: getting signer: getting key from Fulcio: fetching ambient OIDC credentials: invalid character 'u' looking for beginning of value
```
* https://github.com/NVIDIA/cuda-quantum/actions/runs/27628267001/job/81698599772#step:14:26756
```
#74 9.527 [----------] 1 test from qpp_EncodingTester
#74 9.527 [ RUN      ] qpp_EncodingTester.zeroVector
#74 9.527 unknown file: Failure
#74 9.527 Unknown C++ exception thrown in the test body.
```

The first failure looks transient to me, so rerunning makes sense. The second failure looks like an actual test failure that might need to be fixed.

**Comment by 1tnguyen** [comment]
@simsaidan The test failures may be related to `catch (const std::exception &e)`. In the above installer pipeline, we may use a different C++ standard library hence exception catching by a specific type may not be reliable. You can try to do a catch all similar to https://github.com/NVIDIA/cuda-quantum/blob/main/unittests/nvqpp/ptsbe/ExecutePTSBETester.cpp#L42

**Comment by simsaidan** [comment]
@1tnguyen @mitchdz thanks! can we remove this from the queue and retrigger CI if the DCO looks good?

**Comment by 1tnguyen** [comment]
> /ok to test [16fa1e5](https://github.com/NVIDIA/cuda-quantum/pull/4667/commits/16fa1e5bc8620b61d75ec13667848df9859e8ce5)

**Command Bot:** Processing...

**Comment by simsaidan** [comment]
@1tnguyen @sacpis  Do you know why CI is making no progress on those last 4?

**Comment by 1tnguyen** [comment]
> /ok to test [3e52225](https://github.com/NVIDIA/cuda-quantum/pull/4667/commits/3e522256acc641cd9b2613374618e724c481e578)

**Command Bot:** Processing...

**Comment by 1tnguyen** [comment]
> @1tnguyen @sacpis Do you know why CI is making no progress on those last 4?

I've updated the branch with `main` and triggered a new pipeline. Hopefully this will be the one :)

**Comment by simsaidan** [comment]
@1tnguyen Thanks a lot for your help getting this across the line! Could you please assign me to the issue for hackathon tracking purposes?

**Comment by 1tnguyen** [comment]
> @1tnguyen Thanks a lot for your help getting this across the line! Could you please assign me to the issue for hackathon tracking purposes?

I've assigned the issue. Thanks @simsaidan for the contribution.
